### ISSUE TITLE
Expose ScriptLog on CosmosException

### ISSUE THREAD

**Issue by thomaslevesque** [Azure/azure-cosmos-dotnet-v3#1900]
When execution of a stored procedure fails, it would be useful to be able to see the script log. However, even though it's exposed on `StoredProcedureResponse<T>`, it's not exposed on `CosmosException`.

**Describe the solution you'd like**
A `ScriptLog` property on `CosmosException`.

**Describe alternatives you've considered**
The script log is accessible from the headers, but it's not very convenient, because you need to remember the header name, and manually escape the value:

```csharp
var scriptLog = Uri.UnescapeDataString(ex.Headers["x-ms-documentdb-script-log-results"] ?? string.Empty);
```

**Additional context**
N/A

**Comment by j82w** [comment]
I think the best option would be to expose a `CosmosStoredProcedureException : CosmosException` with the additional property. Adding it to the base exception doesn't make since with it only returning for a single API in the SDK. 

@thomaslevesque what is the reason you are using stored procedure instead of [Transactional Batch ](https://devblogs.microsoft.com/cosmosdb/introducing-transactionalbatch-in-the-net-sdk/)or [Bulk](https://devblogs.microsoft.com/cosmosdb/introducing-bulk-support-in-the-net-sdk/)?

**Comment by kirankumarkolli** [comment]
+1 to the concept.

**Comment by thomaslevesque** [comment]
> I think the best option would be to expose a `CosmosStoredProcedureException : CosmosException` with the additional property. Adding it to the base exception doesn't make since with it only returning for a single API in the SDK.

Good point. Yes, a specific exception would work, too. Ideally it would also expose the SubstatusCode returned by the stored procedure (rather than the 400 Bad Request top-level status code)

> what is the reason you are using stored procedure instead of Transactional Batch or Bulk?

Transactional batch or bulk can't do everything a stored procedure can do. For instance, I might need to execute a query and use the result to decide what to do next. And I have stored procedures that don't even perform any write; due to the limitations of the query language, sometimes a stored procedure is the only viable option.

**Comment by thomaslevesque** [comment]
Please reopen. The issue is still relevant, inactivity or not.
